mypy/semanal.py

```diff
@@ -545,7 +545,57 @@ def check_function_signature(self, fdef: FuncItem) -> None:
         elif len(sig.arg_types) > len(fdef.arguments):
             self.fail('Type signature has too many arguments', fdef, blocker=True)
 
+    def check_namedtuple_classdef(self, defn: ClassDef) -> Tuple[List[str], List[Type]]:
+        NAMEDTUP_CLASS_ERROR = 'Invalid statement in NamedTuple definition; ' \
+                               'expected "field_name: field_type"'
+        if self.options.python_version < (3, 6):
+            self.fail('NamedTuple class syntax is only supported in Python 3.6', defn)
+            return [], []
+        if len(defn.base_type_exprs) > 1:
+            self.fail('NamedTuple should be a single base', defn)
+        items = []  # type: List[str]
+        types = []  # type: List[Type]
+        for stmt in defn.defs.body:
+            if isinstance(stmt, AssignmentStmt):
+                if len(stmt.lvalues) > 1 or not isinstance(stmt.lvalues[0], NameExpr):
+                    self.fail(NAMEDTUP_CLASS_ERROR, stmt)
+                    continue
+                name = stmt.lvalues[0].name
+                if not isinstance(stmt.rvalue, TempNode):
+                    # x: int assigns rvalue to TempNode(AnyType())
+                    self.fail('NamedTuple does not support initial values', stmt)
+                if stmt.type is None:
+                    self.fail('NamedTuple field type must be specified', stmt)
```

Comment:
OK. I changed this so that a single "generic" ``... expected "field_name: field_type"`` error is displayed (also for ``x = 1 # type: int``, since this will not work in ``typing.py``). A more specific error ``Right hand side values are not supported in NamedTuple`` is displayed for ``x: int = 1`` (the generic error could be not completely clear in this case).